### PR TITLE
User Guide Documentation Correction

### PR DESCRIPTION
**Pull request by atishamte**
The corrections are made in the form of,
- Spelling correction
- Multiple spaces between words
- Corrected function name
- Use of proper grammar to understand the concept

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPdocs
- [x] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: user_guide_src/source/cli/cli.rst [hunk 1]
```diff
@@ -2,7 +2,7 @@
 Running via the Command Line
 ############################
 
-As well as calling an applications :doc:`Controllers </incoming/controllers>`
+As well as calling an application :doc:`Controllers </incoming/controllers>`
```
Comment (jim-parry) [inline user_guide_src/source/cli/cli.rst]: Should be "application's" (possessive)

File: user_guide_src/source/cli/cli_library.rst [hunk 2]
```diff
@@ -118,14 +118,14 @@ exactly as you would the ``write()`` method::
 
 **wrap()**
 
-This command will take a string, start printing it on the current line, and wrap it to a set length on new lines.
+This command will take a string, start printing it on the current line and wrap it to a set length on new lines.
```
Comment (jim-parry) [inline user_guide_src/source/cli/cli_library.rst]: The removed comma is correct

File: user_guide_src/source/cli/cli_library.rst [hunk 3]
```diff
@@ -118,14 +118,14 @@ exactly as you would the ``write()`` method::
...
 This might be useful when displaying a list of options with descriptions that you want to wrap in the current
 window and not go off screen::
 
 	CLI::color("task1\t", 'yellow');
 	CLI::wrap("Some long description goes here that might be longer than the current window.");
 
-By default the string will wrap at the terminal width. Windows currently doesn't provide a way to determine
+By default, the string will wrap at the terminal width. Windows currently don't provide a way to determine
```
Comment (jim-parry) [inline user_guide_src/source/cli/cli_library.rst]: "doesn't" is correct

File: user_guide_src/source/concepts/mvc.rst [hunk 4]
```diff
@@ -9,15 +9,15 @@ application as separate parts. It should be noted that there are many views on t
 but this document describes our take on it. If you think of it differently, you're free to modify how you use
 each piece as you need.
 
-**Models** manage the data of the application, and help to enforce any special business rules the application might need.
+**Models** manage the data of the application and help to enforce any special business rules the application might need.
 
-**Views** are simple files, with little to no logic, that display the information to the user.
+**Views** are simple files, with little to no logic, that displays the information to the user.
```
Comment (jim-parry) [inline user_guide_src/source/concepts/mvc.rst]: The singular is correct

File: user_guide_src/source/concepts/mvc.rst [hunk 5]
```diff
@@ -69,7 +69,7 @@ Controllers
 ===========
 
 Controllers have a couple of different roles to play. The most obvious one is that they receive input from the user and
-then determine what to do with it. This often involves passing the data to a model to save it, or requesting data from
+then determine what to do with it. This often involves passing the data to a model to save it or requesting data from
```
Comment (jim-parry) [inline user_guide_src/source/concepts/mvc.rst]: Could go either way, but I think it reads better with the comma

File: user_guide_src/source/database/query_builder.rst [hunk 6]
```diff
@@ -581,15 +581,15 @@ Starts a new group by adding an opening parenthesis to the WHERE clause of the q
...
+Ends the current group by adding a closing parenthesis to the WHERE clause of the query.
 
 **************
 Inserting Data
 **************
 
 **$builder->insert()**
 
-Generates an insert string based on the data you supply, and runs the
+Generates an insert string based on the data you supply and runs the
```
Comment (jim-parry) [inline user_guide_src/source/database/query_builder.rst]: Again, couldgo either way, but I think the comma reads better.

File: user_guide_src/source/database/query_builder.rst [hunk 7]
```diff
@@ -653,15 +653,15 @@ will be reset (by default it will be--just like $builder->insert())::
...
+parameter. The reason this worked is that the query has not been executed
 using `$builder->insert()` which resets values or reset directly using
 `$builder->resetQuery()`.
 
 .. note:: This method doesn't work for batched inserts.
 
 **$builder->insertBatch()**
 
-Generates an insert string based on the data you supply, and runs the
+Generates an insert string based on the data you supply and runs the
```
Comment (jim-parry) [inline user_guide_src/source/database/query_builder.rst]: Comma reads better.

File: user_guide_src/source/database/query_builder.rst [hunk 8]
```diff
@@ -833,7 +833,7 @@ performing updates.
 
 **$builder->updateBatch()**
 
-Generates an update string based on the data you supply, and runs the query.
+Generates an update string based on the data you supply and runs the query.
```
Comment (jim-parry) [inline user_guide_src/source/database/query_builder.rst]: Comma reads better.

File: user_guide_src/source/database/transactions.rst [hunk 9]
```diff
@@ -15,10 +15,10 @@ transactions.
 CodeIgniter's Approach to Transactions
 ======================================
 
-CodeIgniter utilizes an approach to transactions that is very similar to
+CodeIgniter utilizes an approach to transactions that are very similar to
```
Comment (jim-parry) [inline user_guide_src/source/database/transactions.rst]: SIngular is cirrect here.

File: user_guide_src/source/database/transactions.rst [hunk 10]
```diff
@@ -15,10 +15,10 @@ transactions.
...
 the process used by the popular database class ADODB. We've chosen that
 approach because it greatly simplifies the process of running
-transactions. In most cases all that is required are two lines of code.
+transactions. In most cases, all that is required are two lines of code.
```
Comment (jim-parry) [inline user_guide_src/source/database/transactions.rst]: Missed "is" two lines of code.

File: user_guide_src/source/extending/events.rst [hunk 11]
```diff
@@ -14,7 +14,7 @@ action when that event is triggered.
 Enabling Events
 ===============
 
-Events are always enabled, and are available globally.
+Events are always enabled and are available globally.
```
Comment (jim-parry) [inline user_guide_src/source/extending/events.rst]: Comma reads better

File: user_guide_src/source/general/errors.rst [hunk 12]
```diff
@@ -15,7 +15,7 @@ keep the best user experience for your users.
 Using Exceptions
 ================
 
-This section is a quick overview for newer programmers, or developers who are not experienced with using exceptions.
+This section is a quick overview for newer programmers or developers who are not experienced with using exceptions.
```
Comment (jim-parry) [inline user_guide_src/source/general/errors.rst]: The comma is better. For clarity, could have "or for developers" too.

File: user_guide_src/source/general/helpers.rst [hunk 13]
```diff
@@ -53,7 +53,7 @@ A helper can be loaded anywhere within your controller methods (or
 even within your View files, although that's not a good practice), as
 long as you load it before you use it. You can load your helpers in your
 controller constructor so that they become available automatically in
-any function, or you can load a helper in a specific function that needs
+any function or you can load a helper in a specific function that needs
```
Comment (jim-parry) [inline user_guide_src/source/general/helpers.rst]: Comma is better

File: user_guide_src/source/general/helpers.rst [hunk 14]
```diff
@@ -106,12 +106,12 @@ with an identical name to the existing Helper.
...
 perhaps add a function or two, or change how a particular helper
 function operates - then it's overkill to replace the entire helper with
-your version. In this case it's better to simply "extend" the Helper.
+your version. In this case, it's better to simply "extend" the Helper.
 
 .. note:: The term "extend" is used loosely since Helper functions are
 	procedural and discrete and cannot be extended in the traditional
 	programmatic sense. Under the hood, this gives you the ability to
-	add to or or to replace the functions a Helper provides.
+	add or replace the functions a Helper provides.
```
Comment (jim-parry) [inline user_guide_src/source/general/helpers.rst]: should be "add to, or to replace"

File: user_guide_src/source/general/logging.rst [hunk 15]
```diff
@@ -39,7 +39,7 @@ the ``/app/Config/Logger.php`` configuration file.
 The ``threshold`` value of the config file determines which levels are logged across your application. If any levels
 are requested to be logged by the application, but the threshold doesn't allow them to log currently, they will be
 ignored. The simplest method to use is to set this value to the minimum level that you want to have logged. For example,
-if you want to log debug messages, and not information messages, you would set the threshold to ``5``. Any log requests with
+if you want to log debug messages and not information messages, you would set the threshold to ``5``. Any log requests with
```
Comment (jim-parry) [inline user_guide_src/source/general/logging.rst]: Original comma is correct

File: user_guide_src/source/general/urls.rst [hunk 16]
```diff
@@ -16,7 +16,7 @@ The segments in the URL, in following with the Model-View-Controller approach, u
 
 1. The first segment represents the controller **class** that should be invoked.
 2. The second segment represents the class **method** that should be called.
-3. The third, and any additional segments, represent the ID and any variables that will be passed to the controller.
+3. The third and any additional segments represent the ID and any variables that will be passed to the controller.
```
Comment (jim-parry) [inline user_guide_src/source/general/urls.rst]: original commas are correct

File: user_guide_src/source/helpers/filesystem_helper.rst [hunk 17]
```diff
@@ -110,7 +110,7 @@ The following functions are available:
 		write_file('./path/to/file.php', $data, 'r+');
 
 	The default mode is 'wb'. Please see the `PHP user guide <http://php.net/manual/en/function.fopen.php>`_
-	for mode options.
+	for more options.
```
Comment (jim-parry) [inline user_guide_src/source/helpers/filesystem_helper.rst]: original "mode" is correct

File: user_guide_src/source/helpers/form_helper.rst [hunk 18]
```diff
@@ -297,9 +297,9 @@ The following functions are available:
 
     	Lets you create a standard drop-down field. The first parameter will
     	contain the name of the field, the second parameter will contain an
-    	associative array of options, and the third parameter will contain the
+    	associative array of options and the third parameter will contain the
```
Comment (jim-parry) [inline user_guide_src/source/helpers/form_helper.rst]: comma is correct

File: user_guide_src/source/helpers/form_helper.rst [hunk 19]
```diff
@@ -297,9 +297,9 @@ The following functions are available:
...
     	value you wish to be selected. You can also pass an array of multiple
-    	items through the third parameter, and the helper will create a
+    	items through the third parameter and the helper will create a
```
Comment (jim-parry) [inline user_guide_src/source/helpers/form_helper.rst]: comma is correct

File: user_guide_src/source/helpers/form_helper.rst [hunk 20]
```diff
@@ -368,7 +368,7 @@ The following functions are available:
 
     	Lets you create a standard multiselect field. The first parameter will
     	contain the name of the field, the second parameter will contain an
-    	associative array of options, and the third parameter will contain the
+    	associative array of options and the third parameter will contain the
```
Comment (jim-parry) [inline user_guide_src/source/helpers/form_helper.rst]: comma is correct

**Comment by jim-parry** [comment]
This is very time-consuming to review. I added comments pointing out where you got carried away with commas, and where some of the typos you corrected were actually not typos, but I only got half-way through. Will continue the review once you have addressed those.

**Comment by atishamte** [comment]
I have used the natural english rules where , & and will only come in a sentence where there are few more comma-separated words in the same line before.
I reverted those changes as you suggested.

**Comment by jim-parry** [comment]
There are so many sets of rules, many of them contradictory :(
We will never please everyone, so should stick to what makes most sense when the text is read, with commas introducing pauses or separating parts of a sentence.
I will continue my review :)

**Comment by atishamte** [comment]
No problem. I will follow those in the future. Cheers...:)

**Comment by jim-parry** [comment]
I have gone through half the changes, and it looks like you flagged the requested changes as "resolved" without making any changes?

**Comment by atishamte** [comment]
> I have gone through half the changes, and it looks like you flagged the requested changes as "resolved" without making any changes?

**Comment by atishamte** [comment]
I have made the changes in the new commit. Please check the files changed panel. I can see those changes.

**Comment by atishamte** [comment]
You can view the changeset here,
https://github.com/codeigniter4/CodeIgniter4/pull/1800/commits/f1aa05bbb27a0be94696e01071f098361a7cd323

File: user_guide_src/source/incoming/message.rst [hunk 21]
```diff
@@ -2,7 +2,7 @@
 HTTP Messages
 =============
 
-The Message class provides an interface to the portions of an HTTP message that are common to both
+The Message class provides an interface to the portions of an HTTP message that is common to both
```
Comment (jim-parry) [inline user_guide_src/source/incoming/message.rst]: "are" is correct. The subject of the relevant sentence fragment: "portions of an HTTP message that **are** ..."

File: user_guide_src/source/incoming/methodspoofing.rst [hunk 22]
```diff
@@ -16,7 +16,7 @@ that you want the request to be::
     </form>
 
 This form is converted into a PUT request and is a true PUT request as far as the routing and the IncomingRequest
-class are concerned.
+class is concerned.
```
Comment (jim-parry) [inline user_guide_src/source/incoming/methodspoofing.rst]: "are" is correct. ... "the routing and the IncomingRequest class **are** ..."

File: user_guide_src/source/incoming/routing.rst [hunk 23]
```diff
@@ -441,7 +441,7 @@ Offsetting the Matched Parameters
 ---------------------------------
 
 You can offset the matched parameters in your route by any numeric value with the ``offset`` option, with the
-value being the number of segments to offset.
+value is the number of segments to offset.
```
Comment (jim-parry) [inline user_guide_src/source/incoming/routing.rst]: "being" is correct

File: user_guide_src/source/libraries/images.rst [hunk 24]
```diff
@@ -261,15 +261,15 @@ Adding a Text Watermark
...
+notices, photographer names, or simply marking the images as a preview so they won't be used in other people's final
 products.
 
 ::
 
 	text(string $text, array $options = [])
 
 The first parameter is the string of text that you wish to display. The second parameter is an array of options
-that allow you to specify how the text should be displayed::
+that allows you to specify how the text should be displayed::
```
Comment (jim-parry) [inline user_guide_src/source/libraries/images.rst]: plural is correct - subject is "options"

File: user_guide_src/source/libraries/pagination.rst [hunk 25]
```diff
@@ -233,9 +233,9 @@ the current page link. The only parameter that it accepts is the number of links
 
 **hasPrevious()** & **hasNext()**
 
-These methods return a boolean true if there are more links than can be displayed on either side of the current page,
+These methods return a boolean true if there are more links that can be displayed on either side of the current page,
```
Comment (jim-parry) [inline user_guide_src/source/libraries/pagination.rst]: "than" is correct

File: user_guide_src/source/libraries/images.rst (hunk 24 above)
Comment (atishamte) [reply user_guide_src/source/libraries/images.rst]: means "allow" is correct?

File: user_guide_src/source/testing/benchmark.rst [hunk 26]
```diff
@@ -22,7 +22,7 @@ With the Timer, you can measure the time between two moments in the execution of
 it simple to measure the performance of different aspects of your application. All measurement is done using
 the ``start()`` and ``stop()`` methods.
 
-The ``start()`` methods takes a single parameter: the name of this timer. You can use any string as the name
+The ``start()`` methods take a single parameter: the name of this timer. You can use any string as the name
```
Comment (jim-parry) [inline user_guide_src/source/testing/benchmark.rst]: should be "method takes"

File: user_guide_src/source/incoming/incomingrequest.rst [hunk 27]
```diff
@@ -154,7 +154,7 @@ This will retrieve data and convert it to an array. Like this::
 **Filtering Input Data**
 
 To maintain security of your application, you will want to filter all input as you access it. You can
-pass the type of filter to use in as the last parameter of any of these methods. The native ``filter_var()``
+pass the type of filter to use it as the last parameter of any of these methods. The native ``filter_var()``
```
Comment (jim-parry) [inline user_guide_src/source/incoming/incomingrequest.rst]: i think this should be "to use as...", so neither in nor it

**Comment by jim-parry** [comment]
Sorry about my earlier message - I thought I deleted it once I realized I was looking at the wrong version.
In any case, I have finished going through the second half too, and added soe comments.

**Comment by atishamte** [comment]
I have made the final changes.

**Pull request merged**